Content.Server/Blob/BlobObserverSystem.cs

```diff
@@ -0,0 +1,862 @@
+
+        foreach (var lookupUid in _lookup.GetEntitiesInRange(position, 5f))
+        {
+            if (!HasComp<BlobTileComponent>(lookupUid))
+                continue;
+            var tileCords = Transform(lookupUid).Coordinates;
+            var distance = Vector2.Distance(position.Position, tileCords.Position);
+
+            if (!(distance < nearestDistance))
+                continue;
+            nearestDistance = distance;
+            nearestEntityUid = lookupUid;
+        }
+
+        return (nearestEntityUid, nearestDistance);
+    }
+
+    private void OnBlobHelp(EntityUid uid, BlobObserverComponent observerComponent,
+        BlobHelpActionEvent args)
+    {
+        _popup.PopupEntity(Loc.GetString("blob-help"), uid, uid, PopupType.Large);
+        args.Handled = true;
+    }
+
+    private void OnSplitCore(EntityUid uid, BlobObserverComponent observerComponent,
+        BlobSplitCoreActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (observerComponent.Core == null || !TryComp<BlobCoreComponent>(observerComponent.Core.Value, out var blobCoreComponent))
+            return;
+
+        if (!blobCoreComponent.CanSplit)
+        {
+            _popup.PopupEntity(Loc.GetString("blob-cant-split"), uid, uid, PopupType.Large);
+            return;
+        }
+
+        var gridUid = args.Target.GetGridUid(EntityManager);
+
+        if (!_map.TryGetGrid(gridUid, out var grid))
+        {
+            return;
+        }
+
+        var centerTile = grid.GetLocalTilesIntersecting(
+            new Box2(args.Target.Position, args.Target.Position)).ToArray();
+
+        EntityUid? blobTile = null;
+
+        foreach (var tileref in centerTile)
+        {
+            foreach (var ent in grid.GetAnchoredEntities(tileref.GridIndices))
+            {
+                if (!TryComp<BlobTileComponent>(ent, out var blobTileComponent))
+                    continue;
+                blobTile = ent;
+                break;
+            }
+        }
+
+        if (blobTile == null || !TryComp<BlobNodeComponent>(blobTile, out var blobNodeComponent))
+        {
+            _popup.PopupEntity(Loc.GetString("blob-target-node-blob-invalid"), uid, uid, PopupType.Large);
+            args.Handled = true;
+            return;
+        }
+
+        if (!_blobCoreSystem.TryUseAbility(uid, observerComponent.Core.Value, blobCoreComponent,
+                blobCoreComponent.SplitCoreCost))
+        {
+            args.Handled = true;
+            return;
+        }
+
+        QueueDel(blobTile.Value);
+        var newCore = EntityManager.SpawnEntity(blobCoreComponent.CoreBlobTile, args.Target);
+        blobCoreComponent.CanSplit = false;
+        if (TryComp<BlobCoreComponent>(newCore, out var newBlobCoreComponent))
+            newBlobCoreComponent.CanSplit = false;
+
+        args.Handled = true;
+    }
+
+
+    private void OnSwapCore(EntityUid uid, BlobObserverComponent observerComponent,
```

Comment:
Core and observer should be doing this cooperatively, instead of the observer outright controlling the core.

It should be possible to wire up the core to an AI in the future for non-player-controlled blob and so far this code isn't great for that.